include/power_grid_model/math_solver/iterative_linear_se_solver.hpp

```diff
@@ -171,8 +188,11 @@ class MeasuredValues {
     }
 
     // power measurement
-    SensorCalcParam<sym> const& bus_injection_power(Idx bus) const {
-        return main_value_[idx_bus_injection_power_[bus]];
+    SensorCalcParam<sym> const& bus_injection(Idx bus) const {
+        return main_value_[bus_injection_[bus].idx_bus_injection];
+    }
+    SensorCalcParam<sym> const& appliance_injection(Idx bus) const {
```

Comment:
I have difficulty to understand why you introduce the concept of `appliance_injection`. It seems to be vague.

If a bus injection if fully measured, we can refer to the `main_value_`, if it is not fully measured, we refer to `partial_injection_`.